github/GitRelease.py

```diff
@@ -175,17 +175,24 @@ def delete_release(self):
             self.url
         )
 
-    def update_release(self, name, message, draft=False, prerelease=False):
+    def update_release(self, name, message, draft=False, prerelease=False, tag_name=None, target_commitish=None):
         """
         :calls: `PATCH /repos/:owner/:repo/releases/:release_id <https://developer.github.com/v3/repos/releases/#edit-a-release>`_
         :rtype: :class:`github.GitRelease.GitRelease`
         """
+        if not tag_name:
+            tag_name = self.tag_name
+        if not target_commitish:
+            target_commitish = self.target_commitish
```

Comment:
I think we should not set `target_commitish` value from `self`, because of this possible scenario:

- You already have an existing release, with an existing `tag_name` and a `target_commitish` at a specific SHA-hash.
- If then this release is edited, and a new `tag_name` that does not exist yet is set and no `target_commitish` is supplied, the behaviour of Github (API) would be to create this new tag with `target_commitish = master`.
- But if we use the value of `self.target_commitish` the result would be that this new tag would be set with `target_commitish = SHA-value`

The right approach would be that if `target_commitish` is not supplied, to not include it in the `post_parameters` at all.

Reply:
The inconsistent naming was also something I noticed, but I assume the maintainer knows why, and whether a refactor would be appropriate.

About editing a release without inputting any `target_commitish`, If you specified its value earlier in the draft-phase, and as long as you don't change the tag_name, the target_commitish will stay the same. I use this in production and it always works like a charm :-)

Reply:
@mfonville If you don't specify commitish when updating a release, and PyGithub doesn't specify it, it sounds like GitHub API will reset commitish to the default branch, at least going off their API documentation, I have not tested it myself. Not sure why it works for you.

Reply:
I use it in production. when creating the draft release I specify the tag_name and the sha-commit to attach it to with target_commitish. When later editing the release I am not changing the tag, so in effect only changing the draft-status. That results in that it retains the target_commitish